keras/src/layers/layer.py

```diff
@@ -1350,10 +1350,11 @@ def __str__(self):
     def __setattr__(self, name, value):
         # Track Variables, Layers, Metrics, SeedGenerators.
         name, value = self._setattr_hook(name, value)
-        if hasattr(self, "_tracker"):
-            value = self._tracker.track(value)
-        elif name != "_tracker":
-            self._initialize_tracker()
+        if name != "_tracker":
+            if not hasattr(self, "_tracker"):
+                self._initialize_tracker()
+            if name != "_build_shapes_dict":
```

Comment:
This is not very generic. I can think of some alternatives:
1. A "do not track" option on a per value basis: `self._build_shapes_dict = tracking.do_not_track(shapes)`
2. A "do not track" option on a per attribute name basis in `Tracker.init()`
3. A way to transform a structure by turning custom collections to plain `dict`s/`list`s. For the better or the worse `optree` and `dmtree` are smart enough to use the same custom classes when doing `map_structure`.

Reply:
Reverted in favor of 3.